### ISSUE TITLE
Could not download FFmpeg

### ISSUE THREAD

**Issue by sioc-de-narf** [cdgriffith/FastFlix#76]
Hi, i tried both  FastFlix_3.0.0_installer.exe and FastFlix_3.0.0_win64.zip and with both i got :

C:\Program Files\FastFlix>FastFlix.exe
2020-10-04 19:19:24 DESKTOP-2OIM4BI fastflix-core[3416] INFO Starting FastFlix 3.0.0
2020-10-04 19:19:25 DESKTOP-2OIM4BI fastflix[14352] DEBUG Using qt engine pyqt5 version 5.15.1
2020-10-04 19:19:25 DESKTOP-2OIM4BI fastflix[14352] DEBUG Using config file "C:\Users\sioc\AppData\Roaming\FastFlix\fastflix.json"
Downloaded 0MB
2020-10-04 19:19:28 DESKTOP-2OIM4BI fastflix[14352] ERROR Could not download FFmpeg
Traceback (most recent call last):
  File "fastflix\gui.py", line 214, in required_info
  File "fastflix\gui.py", line 295, in windows_download_ffmpeg
  File "reusables\file_operations.py", line 80, in extract
TypeError: File is not a known archive
2020-10-04 19:19:28 DESKTOP-2OIM4BI fastflix[14352] ERROR HARD FAIL: Unexpected error: 2
Traceback (most recent call last):
  File "fastflix\gui.py", line 214, in required_info
  File "fastflix\gui.py", line 295, in windows_download_ffmpeg
  File "reusables\file_operations.py", line 80, in extract
TypeError: File is not a known archive

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "fastflix\gui.py", line 263, in start_app
  File "fastflix\gui.py", line 217, in required_info
SystemExit: 2
Unexpected error: 2

**Comment by cdgriffith** [comment]
I apologize I didn't have a chance to respond yet today, thank you for raising the issue! 

As a backup until I look at and fix it you can always download FFmpeg manually: https://www.gyan.dev/ffmpeg/builds/ffmpeg-git-full.zip

Then you will have to update the config file (`fastflix.json` on your system at `C:\Users\sioc\AppData\Roaming\FastFlix\fastflix.json`) to manually point to them.  For example if you created a folder in your user directory called `FFmpeg` and put the binaries (`ffmpeg.exe` and `ffprobe.exe`) from inside the downloaded zip file directly in that folder, the updated config would have these fields: 

```
  "work_dir": "C:\\Users\\sioc\\AppData\\Roaming\\FastFlix",
  "ffmpeg": "C:\\Users\\sioc\\FFmpeg\\ffmpeg.exe",
  "ffprobe": "C:\\Users\\sioc\\FFmpeg\\ffprobe.exe",
```

**Comment by cdgriffith** [comment]
The issue it seems it that `gyan.dev` has updated the name and packaging of FFmpeg to be `7z` archives. 

The "proper" solution is to build FFmpeg myself and bundle it with FFmpeg for windows systems. There are a lot of logistic, server time and legal waters to navigate to do that correctly is why I have been avoiding it. (FFmpeg themselves don't offer builds directly to avoid those issues, and that's why they link to third parties such as `gyan.dev` on their official website.)

**Comment by cdgriffith** [comment]
Fixed in https://github.com/cdgriffith/FastFlix/releases/tag/3.0.2 

Thanks for bringing this to my attention!

**Comment by mediamanrit** [comment]
It appears this is broken again in 3.1.0 .

**Comment by cdgriffith** [comment]
Fixed betterer (I hope) in [3.2.0](https://github.com/cdgriffith/FastFlix/releases/tag/3.2.0)
